spec/controllers/api/sections_controller_spec.rb

```diff
@@ -0,0 +1,23 @@
+describe Api::SectionsController do
+  let(:section) { create :section }
+  let(:instructor) { create :instructor }
+  let(:course) { section.course }
+  context 'An authenticated user requesting' do
+    before :each do
+      instructor.reset_api_key
+      request.env['HTTP_AUTHORIZATION'] = "MarkUsAuth #{instructor.api_key.strip}"
+    end
+    context 'POST create' do
+      it 'should create a new section when given the correct params' do
+        post :create, params: { course_id: course.id, section: { name: 'LEC0301' } }
+        expect(response).to have_http_status(201)
+        expect(Section.find_by(name: 'LEC0301').name).to eq('LEC0301')
```

Comment:
Use `course.sections.find_by` to ensure the section was associated with the correct course